### DISCUSSION TITLE
How to fit a system with only one eclipse

### DISCUSSION THREAD

**Discussion by MinyuLi** [phoebe-project/phoebe2#660] [Questions]
I follow the example of "inverse_paper_examples.py". and when I fit a observational light curve data, which has two eclipses, I got a proper fitting result like this:
![Figure_1](https://user-images.githubusercontent.com/32053311/202710407-217eb330-a9a4-458f-88d5-deacaf661fe5.png)

But when I fit a one eclipse light curve:
![1668775534568](https://user-images.githubusercontent.com/32053311/202710892-a81cecf3-d937-47f5-b7be-d8e07752ceb2.png)

And it seems mask an error secondry eclipse:
![1668775499938](https://user-images.githubusercontent.com/32053311/202711262-62bd41e0-a372-4b0f-a08f-725974170a7e.jpg)

Then I got this error message:
ERROR: primary is overflowing at periastron (requiv=1.0, requiv_max=0.23552918315277127).  Use contact model if overflowing is desired. (4 affected parameters, affecting system,run_compute)
ERROR: secondary is overflowing at periastron (requiv=1.0, requiv_max=0.23552918315277127).  Use contact model if overflowing is desired. (4 affected parameters, affecting system,run_compute)

My question is how can I solve this problem? thanks alot!

**Comment by MinyuLi** [comment]
btw: the phoebe version is 2.3.63

**Comment by kecnry** [comment]
(migrated to a discussion so its easier for others to find and doesn't need to be "closed")

It looks like this is using the `lc_geometry` estimator?  If so - that really is only designed to work (well) for detached EBs with secondary eclipses.  In general, for all estimators, if the results are not reasonable they should not be accepted.  In your case, which looks like a highly-eccentric ellipsoidal variable with a primary eclipse (definitely not something the estimators are likely to handle well), you'll need to manually tweak the parameters to get a physical model that somewhat represents your observations before moving on to optimization and/or sampling techniques.  

I'll make a note to try to improve the documentation to be more clear about what cases the estimators are expected to handle well and how to handle cases where they do not.

**Comment by gecheline** [comment]
To add, the lc_geometry estimator fits Gaussians to what it "thinks" are eclipses and the Gaussian parameters are then used to compute the physical parameters, like radii and temperatures. In your case there's a wide Gaussian fitted to the ellipsoidal variability that is translated into an unreasonably large star which overflows the critical Roche lobe. The simplest solution would be to just manually override the value of the primary radius to avoid that overflow. If you update to phoebe 2.4 you have the option to use another analytical model with lc_geometry - 'polyfit', which might do better here. 

Estimators are designed to provide quick and dirty solutions and each one has limitations. As a general rule, I'd suggest trying out several or all options (see the [2.4 Estimators tutorial](http://phoebe-project.org/docs/2.4/tutorials/LC_estimators)) and choosing the best one, or mix-and-matching the values obtained from more estimators to compensate for their deficiencies if none seem to yield a good result on their own.

**Reply by MinyuLi** [reply]
Thanks for answer, then how to manually override the value of the primary radius ? and I also update phoebe to 2.4.5, how can I use the 'polyfit' model ?

**Reply by kecnry** [reply]
Estimators just return proposed values for a number of parameters, you can then choose to adopt any or all of them and still change other values manually.  So if the rest of the proposed values look reasonable to you, you could adopt those and then manually set a value for the radius (see the tutorials on the website for how to set parameter values).

In 2.4, the `lc_geometry` estimator has an `analytical_model` parameter which can be 'two-gaussian' (default) or 'polyfit'.
